### ISSUE TITLE
Link file formats to existing format registries

### ISSUE THREAD

**Issue by nichtich** [wader/fq#531]
[@wader wrote](https://github.com/wader/fq/issues/529#issuecomment-1357685784):
      
> 👍  I wonder if some formats in fq could have links to these projects? currently format refrences are kept in `<format-name>.md` ex https://github.com/wader/fq/blob/master/format/mp4/mp4.md#references

I think the best registries of file formats to link to are

- [Just Solve the File Format Problem](http://fileformats.archiveteam.org)
- Wikidata (currently with [~14.000 formats](https://w.wiki/68F8))

Both can be edited freely and both link to other registries such as PRONOM. Wikidata further contains more structured information such as default file extension and MIME Type. I once wrote [a Wikidata module for jq](https://github.com/nichtich/jq-wikidata) this may help to get information from Wikidata (e.g. [MP4](https://www.wikidata.org/wiki/Q336316)).

**Comment by nichtich** [comment]
How about another section`### Identifiers` at `<format-name>.md` with a list of URIs. In most cases the Identifier would be a Wikidata URI. The URIs can then be extracted automatically with `fq -d markdown`.

By the way Wikidata also [lists file extensions](https://www.wikidata.org/wiki/Property:P1195), these can be harvested and compared from there.

P.S.: I started [in this branch](https://github.com/wader/fq/compare/master...nichtich:fq:issue-531), does this make sense?

**Comment by wader** [comment]
> How about another section`### Identifiers` at `<format-name>.md` with a list of URIs. In most cases the Identifier would be a Wikidata URI. The URIs can then be extracted automatically with `fq -d markdown`.
> 
> By the way Wikidata also [lists file extensions](https://www.wikidata.org/wiki/Property:P1195), these can be harvested and compared from there.

Either own section or having them under references? maybe make the link titles a but more descriptive?  maybe something like:

- [Wikidata - Avro Object Container File (Q105855052)](http://www.wikidata.org/entity/Q105855052)
- [National Archives - Apache Avro  (fmt/2029)](https://www.nationalarchives.gov.uk/PRONOM/fmt/2029)

BTW `fq -d markdown` uses the md markdown to output the help so it should just work ™️ 

> P.S.: I started [in this branch](https://github.com/wader/fq/compare/master...nichtich:fq:issue-531), does this make sense?

I think so. Feel like continue working on it?

**Comment by nichtich** [comment]
Having them under references seems more useable for human reader and the mappings can be extracted anyway. I'll do a pull request when finished the work.

**Comment by wader** [comment]
👍 what would be the main use case? educational, quickly find spec and related formats etc?

**Comment by nichtich** [comment]
I manage [a catalogue of data formats](https://format.gbv.de/) and have a project that involves [locating errors in documents of arbitrary formats](https://gbv.github.io/data-validation-report-format/#sec-additional-dimensions). Not having to manage a list of formats (and their short name) supported by fq would be helpful for long-term support. Still this is no urgent request, it will likely take some weeks.

**Comment by wader** [comment]
I see, no stress. Is there an implantation of a validator somewhere?

**Comment by nichtich** [comment]
> I see, no stress. Is there an implantation of a validator somewhere?

No, It's only a minor part of an [ongoing research project](https://doi.org/10.5281/zenodo.18430881) to facilitate managing data quality requirements. Users can choose a template such as

> $X should be between $Y and $Z

And insert parameters such as

> "[Age of person] should be between [18] and [32]"
> "[Image width in pixel] should be between [200] and [800]"

Some parameters refer to elements from a data format, each with a human readable label and a locator how to get the element from a document, e.g.

- Age of person = `//person/@age` (XPath)
- Image width in pixel = `png:.chunks[0].width` (fq)

For XML documents we just need an XPath engine to refer to data elements. For binary formats there was no generic method to access data elements before `fq`.

**Comment by wader** [comment]
👍 one feedback for fq usage might be to that a user should try write "robust" filters. Ex things like `.chunks[] | select(.type == "IHDR").width` but in the case of PNG i guess IHDR is required to be first.
